pkgs/by-name/op/opencode/package.nix

```
@@ -147,6 +147,13 @@ stdenvNoCC.mkDerivation (finalAttrs: {
     mkdir -p $out/bin
     install -Dm755 opencode $out/bin/opencode
 
+    ${lib.optionalString (stdenvNoCC.buildPlatform.canExecute stdenvNoCC.hostPlatform) ''
```

According to https://nixos.org/manual/nixpkgs/stable/#installshellfiles-installshellcompletion you need `installShellFiles` in `nativeBuildInputs`. When I look at some other packages (for example [`just`](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/by-name/ju/just/package.nix)) they do the following which is pretty neat.

Add this as input:

```nix
{
...
installShellFiles
installShellCompletions ? stdenvNoCC.buildPlatform.canExecute stdenvNoCC.hostPlatform,
}:
```
Then add `installShellFiles` to the inputs if `installShellCompletions` is `true`. In our case:

```nix
nativeBuildInputs = [
  bun
  writableTmpDirAsHomeHook
] ++ lib.optional (installShellCompletions) installShellFiles;
```

And finally:

```suggestion
    ${lib.optionalString installShellCompletions ''
```